src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -3,12 +3,94 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import java.util.Objects;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private final int minAge = 18;
+    private final int minLength = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws InvalidDataException {
+        somethingEqualsNull(user);
+        ageIsCorrect(user);
+        correctLogin(user);
+        shortPassword(user);
+        duplicate(user);
+        passwordWithoutSymbols(user);
+        passwordWithoutUpperCaseLetters(user);
```

Comment:
```suggestion
        checkUser(user)
        checkAge(user);
        checkLogin(user);
        checkPassword(user);
```

Reply:
In this methods check for null value at first, after this check is valid length or any